dRAGon/search.py

```diff
@@ -0,0 +1,68 @@
+import whoosh
+import os.path
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+from whoosh.index import create_in
+from whoosh.fields import * # TODO: Get rid of this for the love of god and all attendant bodhisattvas
+from whoosh.qparser import QueryParser
+from whoosh.index import open_dir
+
+
+## Define search method
+
+def search_rulebook(query=""):
+    ix = open_dir("indexdir")
+    with ix.searcher() as searcher:
+        query = QueryParser("content", ix.schema).parse(query)
+        results = searcher.search(query)
+        return results[0]
+
+
+
+
+## Initialize chunkers
+
+text_splitter = RecursiveCharacterTextSplitter(
+        # Chunk params
+        chunk_size=300,
+        chunk_overlap=50,
+        length_function=len, #We are counting characters, not tokens
+        is_separator_regex=False,
+)
+
+
+
+
+
+## Index Content
+
+
+#schema = Schema(title=TEXT(stored=True), path=ID(stored=True), content = TEXT)
+schema = Schema(id_=ID(stored=True), content = TEXT(stored=True))
+
+# create index if it does not exist
+if not os.path.exists("indexdir"):
+    # Load data
+    with open("../../cairn-markdown.md") as file:
+        rulebook = file.read()
+    # Chunk
+
+    chunks = text_splitter.split_text(rulebook)
+
+    os.mkdir("indexdir")
```

Comment:
if you need to use a real directory we should put something in the OS temp directory for dragon imo

Reply:
Also, in the future could this be enhanced to use a database instead of a document? Or to have the option of a database

Reply:
also also: can we cleanup the index on exit? might be helpful to make this whole thing into a class/context manager?

Reply:
Unless it's very _very_ expensive to build the index, then maybe we should leave it..